js/collections/Listings.js

```diff
@@ -41,4 +41,19 @@ export default class extends Collection {
     // https://github.com/OpenBazaar/openbazaar-go/issues/143
     return cats.sort();
   }
+
+  /**
+   * Returns a list of the aggregate listing types from all of
+   * the listings in the collection.
+   */
+  get types() {
+    const types = [];
+
+    this.models.forEach(listing => {
+      const type = listing.get('contractType');
+      if (types.indexOf(type) === -1) types.push(type);
+    });
+
+    return types.sort();
```

Comment:
That's pretty nifty Josh.

One minor thing though is I think it could do without the concat:

`return [...new Set(this.pluck('categories'))].sort()`

Reply:
I think you need the concat because each categories is an array, without it you get a set of arrays.

Reply:
oh right, didn't realize those were arrays.

But you still need another spread operator on the Set which converts the return value to an array. Also, the position of the sort needs to slightly move so it operates on the result of the concat.

This should work:

`return [...new Set([].concat(...this.pluck('categories')).sort())]`

Reply:
either way, it's a pretty clever one-liner!